--- a/src/video/kmsdrm/SDL_kmsdrmvideo.h	2022-12-17 13:11:20.000000000 +0300
+++ b/src/video/kmsdrm/SDL_kmsdrmvideo.h	2023-05-25 11:26:28.645427448 +0300
@@ -33,8 +33,10 @@
 #include <xf86drm.h>
 #include <xf86drm.h>
 #include <xf86drmMode.h>
+#if SDL_VIDEO_OPENGL_EGL
 #include <gbm.h>
 #include <EGL/egl.h>
+#endif
 
 typedef struct SDL_VideoData
 {
@@ -117,7 +119,9 @@
     SDL_bool waiting_for_flip;
     SDL_bool double_buffer;
 
+#if SDL_VIDEO_OPENGL_EGL
     EGLSurface egl_surface;
+#endif
     SDL_bool egl_surface_dirty;
 } SDL_WindowData;
 
