--- a/include/configs/sun8i.h	2023-04-29 21:29:53.480410277 +0300
+++ b/include/configs/sun8i.h	2023-04-29 21:37:59.881389822 +0300
@@ -27,6 +27,13 @@
 	#define CONFIG_SUNXI_USB_PHYS	2
 #endif
 
+#define CONFIG_BOOTCOMMAND "setenv bootm_boot_mode sec; " \ 
+                            "load mmc 0:1 0x41000000 zImage; " \
+                            "load mmc 0:1 0x41800000 sun8i-v3s-licheepi-zero-dock.dtb; " \
+                            "bootz 0x41000000 - 0x41800000;"
+
+#define CONFIG_BOOTARGS "console=tty0 panic=5 rootwait root=/dev/mmcblk0p2 earlyprintk rw vt.global_cursor_default=0"
+
 /*
  * Include common sunxi configuration where most the settings are
  */
